sdk/src/main/java/com/horizen/transaction/OpenStakeTransaction.java

```diff
@@ -0,0 +1,184 @@
+package com.horizen.transaction;
+
+import com.google.common.primitives.Ints;
+import com.horizen.box.BoxUnlocker;
+import com.horizen.box.ZenBox;
+import com.horizen.box.data.ZenBoxData;
+import com.horizen.proof.Proof;
+import com.horizen.proposition.Proposition;
+import com.horizen.proposition.PublicKey25519Proposition;
+import com.horizen.secret.PrivateKey25519;
+import com.horizen.secret.Secret;
+import com.horizen.transaction.exception.TransactionSemanticValidityException;
+import com.horizen.utils.Pair;
+import scala.Array;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+
+import static com.horizen.transaction.CoreTransactionsIdsEnum.OpenStakeTransactionId;
+
+public class OpenStakeTransaction extends SidechainNoncedTransaction<PublicKey25519Proposition, ZenBox, ZenBoxData>{
+    public final static byte OPEN_STAKE_TRANSACTION_VERSION = 1;
+
+    final List<byte[]> inputsIds;
+    private final List<ZenBoxData> outputsData;
+    final List<Proof<Proposition>> proofs;
+
+    private final long fee;
+
+    private final byte version;
+
+    private List<BoxUnlocker<PublicKey25519Proposition>> unlockers;
+
+    private final int forgerListIndex;
+
+    public OpenStakeTransaction(List<byte[]> inputsIds,
+                                List<ZenBoxData> outputsData,
+                                List<Proof<Proposition>> proofs,
+                                int forgerListIndex,
+                                long fee,
+                                byte version) {
+        Objects.requireNonNull(inputsIds, "Inputs Ids list can't be null.");
```

Comment:
Consider enforcing the condition to have `inputsIds.size() == 1` here, to catch it earlier (it is also verified by the `transactionSemanticValidity()` method)